File: microservices/resume-service/src/Domain/Entities/Resume.cs
using SharedKernel;

namespace Domain.Entities;
public class Resume : Entity
{
    public string Id { get; set; }
    public string UserId { get; set; }
    public string Name { get; set; }
    public string UserInfo { get; set; }
    public List<ProfileEntry> ProfileEntries { get; set; }
    public string ResumeInfo { get; set; }
    public string Keywords { get; set; }
    public string? JobPosting {  get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime ModifiedAt { get; set; }

    public Resume()
    {
        Id = Guid.NewGuid().ToString();
    }
}

File: frontend/job-application-helper-ui/src/app/resume/shared/models/resume.ts
import { HelperMethods } from "../../../core/helper-methods";
import { ProfileEntry } from "../../../profile-entry/shared/models/profile-entry";
import { ResumeInfo } from "./resume-info";
import { UserPersonalInfo } from "./user-personal-info";

export class Resume
{
    id: string;
    name: string;
    userInfo: UserPersonalInfo;
    profileEntries: ProfileEntry[];
    resumeInfo: ResumeInfo;
    keywords: string[];
    jobPosting: string | null = null;
    createdAt: Date;
    modifiedAt: Date;

    constructor()
    {
        this.id = '';
        this.name = '';
        this.userInfo = {
            fullName: '',
            contactInfo: []
        };
        this.createdAt = new Date();
        this.modifiedAt = new Date();
        this.profileEntries = [];
        this.keywords = [];
        this.resumeInfo = {
            templateId: '',
            sections: []
        };
    }    
}